### PR TITLE
ChangeFeed: Adds helpful error message for 304 exceptions.

### PR DESCRIPTION
**Pull request by bchong95**
# ChangeFeed: Adds helpful error message for 304 exceptions.

## Description

Adds an actual error message for ChangeFeed 304 Not Modified Exceptions.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/ChangeFeed/ChangeFeedIteratorCore.cs [hunk 1]
```diff
@@ -205,7 +205,9 @@ public override async Task<ResponseMessage> ReadNextAsync(CancellationToken canc
             }
             else
             {
-                responseMessage = new ResponseMessage(statusCode: System.Net.HttpStatusCode.NotModified);
+                responseMessage = new ResponseMessage(
+                    statusCode: System.Net.HttpStatusCode.NotModified, 
+                    errorMessage: "No new changes from the server for ChangeFeed. Learn More @ https://aka.ms/changefeed304");
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/ChangeFeed/ChangeFeedIteratorCore.cs:210]: Worth adding to the Resources files?

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/ChangeFeed/ChangeFeedIteratorCore.cs:210]: This should be updated in master. Also the aka.ms link should follow the format of the other tsg links.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/ChangeFeed/ChangeFeedIteratorCore.cs:210]: There isn't much point to the resource files if we don't maintain the other languages. 

This is also a client side generated error message, so updating in master doesn't make much sense.

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/ChangeFeed/ChangeFeedIteratorCore.cs:210]: The reason to update master is that legacy gateway and the v2 SDK get the link too. We should still use the resource file in-case in the future we are required to start maintaining multiple languages.

**Review by ealsur — APPROVED** [review]

**Comment by j82w** [comment]
Closing since there has been no update for over 30 days. Please re-open if necessary.

**Pull request closed without merging**